include/mbedtls/check_config.h

```diff
@@ -753,10 +753,9 @@
  * Otherwise support for at least one ciphersuite mandates either SHA_256 or
  * SHA_384.
  */
-#if defined(MBEDTLS_SSL_PROTO_TLS1_3) && \
-    ( ( !defined(MBEDTLS_HKDF_C) ) || \
-      ( !defined(MBEDTLS_SHA256_C) && !defined(MBEDTLS_SHA384_C) ) || \
-      ( !defined(MBEDTLS_PSA_CRYPTO_C) ) )
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3) &&                   \
```

Comment:
We need to require `PSA_WANT_ALG_HKDF_EXTRACT` and `PSA_WANT_ALG_HKDF_EXPAND`. Also, I we need to keep `MBEDTLS_PSA_CRYPTO_C` too (TLS 1.3 unconditionally requires it).

I'm wondering it it might be worth splitting this, and the comment above, into:
```
/* TLS 1.3 requires separate HKDF parts from PSA */
#if defined(MBEDTLS_SSL_PROTO_TLS1_3) && ! ( PSA + both HKDF-xxx )
#error "..."
#endif

/* TLS 1.3 requires at least one ciphersuite, so at least SHA-256 or SHA-384 */
// the big condition about hashes + #error
```
Wdyt?

Reply:
I think it is better option then expanding existing conditions for TLS 1.3.

Reply:
I'd like to suggest the following split:
```c
/* TLS 1.3 requires at least one ciphersuite, so at least SHA-256 or SHA-384 */
#if defined(MBEDTLS_SSL_PROTO_TLS1_3)
/* We always need at least one of the hashes via PSA (for use with HKDF) */
#if !( defined(PSA_WANT_ALG_SHA_256) || defined(PSA_WANT_ALG_SHA_348) )
#error "MBEDTLS_SSL_PROTO_TLS1_3 defined, but not all prerequisites"
#endif
#if !defined(MBEDTLS_USE_PSA_CRYPTO)
/* When USE_PSA_CRYPTO is not defined, we also need SHA-256 or SHA-384 via the
 * legacy interface, including via the MD layer, for the parts of the code
 * that are shared with TLS 1.2 (running handshake hash). */
#if !defined(MBEDTLS_MD_C) || \
    !( defined(MBEDTLS_SHA256_C) || defined(MBEDTLS_SHA384_C) )
#error "MBEDTLS_SSL_PROTO_TLS1_3 defined, but not all prerequisites"
#endif
#endif /* !USE_PSA */
#endif /* TLS 1.3 */
```
IMO this is much more readable than the current boolean expression with 4 levels of parentheses (which I'm aware I originally suggested, but now regret).